### PR TITLE
Bb more complex blackout dates

### PR DESCRIPTION
**Pull request by breanneboland**
## Description

This is an additional small change toward the goal of taking fields beyond dates into consideration when saying if a blackout date applies to a move or not. This PR changes one argument for all blackout date querying functions from PickupDate (time.Time) to a ShipmentWithOffer object. 

## Verification Steps

* [x] All tests pass.
* [x] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely) have been satisfied.

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/155643358) for this change

### CODE REVIEW

File: pkg/models/blackout_dates.go [hunk 1]
```diff
@@ -19,14 +19,14 @@ type BlackoutDate struct {
...
+	SourceGBLOC                     *string    `json:"source_gbloc" db:"source_gbloc"`
 	Zip3                            *int       `json:"zip3" db:"zip3"`
 	VolumeMove                      *bool      `json:"volume_move" db:"volume_move"`
 }
 
 // FetchTSPBlackoutDates runs a SQL query to find all blackout_date records connected to a TSP ID.
-func FetchTSPBlackoutDates(tx *pop.Connection, tspID uuid.UUID) ([]BlackoutDate, error) {
+func FetchTSPBlackoutDates(tx *pop.Connection, tspID uuid.UUID, shipment ShipmentWithOffer) ([]BlackoutDate, error) {
 	blackoutDates := []BlackoutDate{}
 	// TODO: update query to do the work of seeing if the proposed pickup date being checked
```
[the remark this answers is not in the document]
Comment (breanneboland) [reply pkg/models/blackout_dates.go]: D'oh, I totally did on one iteration of this, but clearly not this one. Thank you.

File: pkg/testdatagen/make_blackout_date_records.go [hunk 2]
```diff
@@ -79,17 +79,17 @@ func MakeBlackoutDateData(db *pop.Connection) {
...
-	// Make a blackout date with cos, market, gbloc, and channel.
+	// Make a blackout date with market and source gbloc.
 	MakeBlackoutDate(db,
 		tspList[rand.Intn(len(tspList))],
 		time.Now(),
 		time.Now(),
 		&tdlList[rand.Intn(len(tdlList))],
-		&gbloc,
 		&market,
+		&gbloc,
```
[the remark this answers is not in the document]
Comment (breanneboland) [reply pkg/testdatagen/make_blackout_date_records.go]: INDEED.

**Comment by breanneboland** [comment]
Nice! Thanks, @kilbergr.

File: pkg/awardqueue/awardqueue_test.go [hunk 3]
```diff
@@ -107,11 +107,38 @@ func (suite *AwardQueueSuite) Test_ShipmentWithinBlackoutDates() {
...
+
 	// One TSP with no blackout dates
 	testTSP2, _ := testdatagen.MakeTSP(suite.db, "A Spotless TSP", "PORK")
 
+	// Two shipments with offers, using the shipments above
+	shipmentWithOfferBetween := models.ShipmentWithOffer{
+		ID: testShipmentBetween.ID,
+		TrafficDistributionListID:       testTDL.ID,
+		PickupDate:                      testPickupDateBetween,
+		TransportationServiceProviderID: nil,
```
Comment (jim) [inline pkg/awardqueue/awardqueue_test.go]: These `nil` fields can be left out of the struct literal if it makes the code cleaner. I don't think you necessarily should to do that here, and sometimes it is useful to be explicit about the `nil` values.

Comment (breanneboland) [reply pkg/awardqueue/awardqueue_test.go]: Makes sense. It'd be useful to keep talking about when it adds clarity and when it adds unnecessary length. For now - related to that hard-coded value we discovered/rediscovered yesterday, I'm all about the values being suuuuuper explicit.

**Pull request merged**